### PR TITLE
codeintel: Add a feature flag for auto indexing

### PR DESCRIPTION
**Pull request by efritz**
Add feature flag in site settings `codeIntelAutoIndexing.enabled` which will enable or disable code intel index routes, background routines, resolvers, and UI navigation related to auto indexing.

This subsumes an envvar that enabled or disabled background routines.

### CODE REVIEW

File: client/web/src/enterprise/site-admin/sidebaritems.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ const codeIntelGroup: SiteAdminSideBarGroup = {
         {
             to: '/site-admin/code-intelligence/indexes',
             label: 'Auto indexing',
-            condition: () => Boolean(window.context?.sourcegraphDotComMode),
+            condition: () => Boolean(window.context?.autoIndexingEnabled),
```
Comment (eseliger) [inline client/web/src/enterprise/site-admin/sidebaritems.ts:48]: just realized: wouldn't the routes need to have a changed condition as well? Or are they always accessible in enterprise, regardless of the flag? I think that could be confusing when using the browser history or bookmarks.

**Review by eseliger — APPROVED** [review]
🌟

File: client/web/src/enterprise/site-admin/sidebaritems.ts (hunk 1 above)
Comment (efritz) [reply client/web/src/enterprise/site-admin/sidebaritems.ts:48]: Should probably be changed as well.

**Review by uwedeportivo — APPROVED** [review]

File: schema/schema.go [hunk 2]
```diff
@@ -1175,6 +1175,8 @@ type SiteConfiguration struct {
 	CampaignsReadAccessEnabled *bool `json:"campaigns.readAccess.enabled,omitempty"`
 	// CampaignsRestrictToAdmins description: When enabled, only site admins can create and apply campaigns.
 	CampaignsRestrictToAdmins bool `json:"campaigns.restrictToAdmins,omitempty"`
+	// CodeIntelAutoIndexingEnabled description: Enables/disables the code intel auto indexing feature.
+	CodeIntelAutoIndexingEnabled *bool `json:"codeIntelAutoIndexing.enabled,omitempty"`
```
Comment (daxmc99) [inline schema/schema.go:1179]: Why is this a pointer bool? Seems like we could pull down to false when not enabled

**Review by daxmc99 — APPROVED** [review]

File: schema/schema.go (hunk 2 above)
Comment (efritz) [reply schema/schema.go:1179]: This is auto-generated 🤷

**Pull request merged**